TileFusionManager.java

```diff
@@ -0,0 +1,234 @@
+                        gameBoard[newRow][col] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * Moves all tiles left and merges adjacent tiles with the same value.
+     */
+    private void moveLeft() {
+        for (int row = 0; row < gridSize; row++) {
+            for (int col = 1; col < gridSize; col++) {
+                if (gameBoard[row][col] != 0) {
+                    int newCol = col;
+                    while (newCol > 0 && gameBoard[row][newCol - 1] == 0) {
+                        gameBoard[row][newCol - 1] = gameBoard[row][newCol];
+                        gameBoard[row][newCol] = 0;
+                        newCol--;
+                        hasMoved = true;
+                    }
+                    if (newCol > 0 && gameBoard[row][newCol - 1] == gameBoard[row][newCol]) {
+                        gameBoard[row][newCol - 1] *= 2;
+                        currentScore += gameBoard[row][newCol - 1];
+                        gameBoard[row][newCol] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * Moves all tiles right and merges adjacent tiles with the same value.
+     */
+    private void moveRight() {
+        for (int row = 0; row < gridSize; row++) {
+            for (int col = gridSize - 2; col >= 0; col--) {
+                if (gameBoard[row][col] != 0) {
+                    int newCol = col;
+                    while (newCol < gridSize - 1 && gameBoard[row][newCol + 1] == 0) {
+                        gameBoard[row][newCol + 1] = gameBoard[row][newCol];
+                        gameBoard[row][newCol] = 0;
+                        newCol++;
+                        hasMoved = true;
+                    }
+                    if (newCol < gridSize - 1 && gameBoard[row][newCol + 1] == gameBoard[row][newCol]) {
+                        gameBoard[row][newCol + 1] *= 2;
+                        currentScore += gameBoard[row][newCol + 1];
+                        gameBoard[row][newCol] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * Adds a new tile (with value 2) to a random empty cell.
+     */
+    private void addNewTile() {
+        for (int i = 0; i < gridSize; i++) {
+            for (int j = 0; j < gridSize; j++) {
+                if (gameBoard[i][j] == 0) {
+                    gameBoard[i][j] = 2;
+                    return;
+                }
+            }
+        }
+    }
+    /**
+     * Resets the game state to start a new game.
+     */
+    private void resetGame() {
+        gameBoard = new int[gridSize][gridSize];
+        currentScore = 0;
+        // Initialize the board with two starting tiles
+        gameBoard[0][0] = 2;
+        gameBoard[1][1] = 2;
+        tileCanvas.updateBoard(gameBoard, currentScore);
+    }
+}

```

Comment:
Do not use documentation for the methods whose name describes the method itself.